ui/app/pages/confirmation/confirmation.js

```diff
@@ -0,0 +1,243 @@
+
+  // Generating templatedValues is potentially expensive, and if done on every render
+  // will result in a new object. Avoiding calling this generation unnecessarily will
+  // improve performance and prevent unnecessary draws.
+  const templatedValues = useMemo(() => {
+    return pendingConfirmation
+      ? getTemplateValues(pendingConfirmation, t, dispatch)
+      : {};
+  }, [pendingConfirmation, t, dispatch]);
+
+  useEffect(() => {
+    const environmentType = getEnvironmentType();
+    // If the number of pending confirmations reduces to zero when the user
+    // is in the fullscreen or popup UI, return them to the default route.
+    // Otherwise, if the number of pending confirmations reduces to a number
+    // that is less than the currently viewed index, reset the index.
+    if (
+      pendingConfirmations.length === 0 &&
+      (environmentType === ENVIRONMENT_TYPE_FULLSCREEN ||
+        environmentType === ENVIRONMENT_TYPE_POPUP)
+    ) {
+      history.push(DEFAULT_ROUTE);
+    } else if (pendingConfirmations.length <= currentPendingConfirmation) {
+      setCurrentPendingConfirmation(pendingConfirmations.length - 1);
+    }
+  }, [pendingConfirmations, history, currentPendingConfirmation]);
+  if (!pendingConfirmation) {
+    return null;
+  }
+
+  return (
+    <div className="confirmation-page">
+      {pendingConfirmations.length > 1 && (
+        <div className="confirmation-page__navigation">
+          <p>
+            {currentPendingConfirmation + 1} of {pendingConfirmations.length}{' '}
+            pending
+          </p>
+          {currentPendingConfirmation > 0 && (
+            <button
+              className="confirmation-page__navigation-button"
+              onClick={() =>
+                setCurrentPendingConfirmation(currentPendingConfirmation - 1)
+              }
+            >
+              <i className="fas fa-chevron-left"></i>
+            </button>
+          )}
+          <button
+            className="confirmation-page__navigation-button"
+            disabled={
+              currentPendingConfirmation + 1 === pendingConfirmations.length
+            }
+            onClick={() =>
+              setCurrentPendingConfirmation(currentPendingConfirmation + 1)
+            }
+          >
+            <i className="fas fa-chevron-right"></i>
+          </button>
+        </div>
+      )}
+      <div className="confirmation-page__content">
+        <Box justifyContent="center">
+          <NetworkDisplay
+            colored={false}
+            indicatorSize={SIZES.XS}
+            labelProps={{ color: COLORS.BLACK }}
+          />
+        </Box>
+        <Box justifyContent="center" padding={[1, 4, 4]}>
+          <Chip
+            label={stripHttpSchemes(originMetadata.origin)}
+            labelColor="gray"
+            leftIcon={
+              <SiteIcon
+                icon={originMetadata.icon}
+                iconName={originMetadata.name}
```

Comment:
Where in the designs are you seeing this? It's only used as a fallback when the icon isn't supplied. I'm not questioning you just can't find it to confirm the right behavior and resolve.

Reply:
Sorry, when I said "according to the latest designs" I meant that the `name` was _absent_ from all of our _current_ Figma designs. I don't believe it has been documented anywhere that it should not be used. Though this has been touched on once or twice recently during design syncs I think.

What I recall about the site icon from the 'LoginPerSite" designs is that the fallback icon name should be the hostname. Even for HTTP, so that the fallback icon isn't "H" for everything. But again I'm unsure whether this is documented anywhere unfortunately.

The basically reason we had for not using `originMetadata.name` is that we noticed that in practice it was full of junk. We get it from the site title, and dapps tend to put all sorts of nonsense there, not just their names.

Reply:
Thanks! so just to recap:
- [ ] name should just be hostname in useOriginMetadata (your suggestion that I can just commit now)
- [ ] the name prop on siteIcon should use hostname.
- [ ] change this behavior anywhere <SiteIcon> is consuming origin Metadata to use hostname instead?

Reply:
Sounds right to me, yep!